src/runtime/pkg/utils/schedcore.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022 Apple Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package utils
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+// PidType is the type of provided pid value and how it should be treated
+type PidType int
+
+const (
+	// Pid affects the current pid
+	Pid PidType = pidtypePid
+	// ThreadGroup affects all threads in the group
+	ThreadGroup PidType = pidtypeTgid
+	// ProcessGroup affects all processes in the group
+	ProcessGroup PidType = pidtypePgid
+)
+
+const (
+	pidtypePid  = 0
+	pidtypeTgid = 1
+	pidtypePgid = 2
```

Comment:
- Could you define these values before they are used for easier reading?
- Could you expand these two names? Pid is clear enough but `Tg` and `Pg` are not very clear. Why not just use `pidTypeThreadGroup` and `pidTypeProcessGroup`.